extensions/amp-sidebar/0.1/toolbar.js

```diff
@@ -82,21 +82,18 @@ export class Toolbar {
       this.toolbarDomElement_
     );
     // Set the target element to the toolbar clone if it exists.
-    this.ampdoc_.whenReady().then(() => {
-      const targetElement = this.ampdoc_.getElementById(targetId);
-      if (targetElement) {
-        this.toolbarTarget_ = targetElement;
-        this.toolbarClone_.classList.add('i-amphtml-toolbar');
-        toggle(this.toolbarTarget_, false);
-      } else {
-        // This error will be later rethrown as a user error and
-        // the side bar will continue to function w/o toolbar feature
-        throw new Error(
-          'Could not find the ' +
-            `toolbar-target element with an id: ${targetId}`
-        );
-      }
-    });
+    const targetElement = this.ampdoc_.getElementById(targetId);
+    if (targetElement) {
+      this.toolbarTarget_ = targetElement;
+      this.toolbarClone_.classList.add('i-amphtml-toolbar');
+      toggle(this.toolbarTarget_, false);
+    } else {
+      // This error will be later rethrown as a user error and
+      // the side bar will continue to function w/o toolbar feature
+      throw new user().createError(
```

Comment:
The comment describes throwing a non-blocking error, which I don't think this was doing before. I am also not sure what "later rethrown" is referring to. Does `createError` ensure the behavior described? I recommend either using `user().warn()` to be able to "continue to function" as described, or placing this as a `userAssert` and removing the legacy comment.

Reply:
I think "later rethrown" refers to the try/catch in amp-sidebar. I'll make this comment less confusing.

Reply:
Thanks. Also looks like you want to delete `new` here and in the v2 file.

Reply:
Donezo.